### ISSUE TITLE
acquireTokenSilent() doesn't get new ID token

### ISSUE THREAD

**Issue by navzam** [AzureAD/microsoft-authentication-library-for-js#1241]
## Library
- [x] `msal@1.2.0`
- [ ] `@azure/msal-browser@2.x.x`
- [ ] `@azure/msal-angular@0.x.x`
- [ ] `@azure/msal-angular@1.x.x`
- [ ] `@azure/msal-angularjs@1.x.x`

## Description
If I call `acquireTokenSilent()` with the `openid` scope after the cached token has expired, the result contains a new access token and the expired ID token. I don't see a new ID token anywhere in the result.

The request to the `/authorize` endpoint has `response_type=token` (not `response_type=id_token token`) so I assume AAD isn't giving back an ID token.

## Regression
Not tested on other versions of MSAL.

## Configuration
MSAL configuration options:

```js
{
  auth: {
    clientId: AAD_APP_ID,
    redirectUri: AAD_REDIRECT_URI,
  },
  cache: {
    storeAuthStateInCookie: true,
  },
}
```

## Reproduction steps
1. Call `acquireTokenSilent()` for the first time:

```js
let authTokenResult = await myMSALObj.acquireTokenSilent({
  scopes: ['user.read', 'openid'],
  loginHint: USER_UPN,
});

// Take note of authTokenResult.idToken.rawIdToken and authTokenResult.idToken.expiration
```

2. Wait for the cached access and ID tokens to expire (usually > 1 hr).
3. Call `acquireTokenSilent()` again with the same parameters.
4. Observe that `authTokenResult.idToken.rawIdToken` and `authTokenResult.idToken.expiration` are the same as before, and there is no new ID token.

## Expected behavior
I expected to have a valid ID token after calling `acquireTokenSilent()` with the `openid` scope.

## Browsers
Tested on Edge (81.0.396.0) and Chrome (79.0.3945.130)

**Comment by arsnyder16** [comment]
Seeing this issue as well. As a workaround for now i am always passing ` forceRefresh: true` when i need an `idToken`

**Comment by arunprasathv** [comment]
How do you call acquireTokenSilent when token expired. Do you call it from failure() ?

**Comment by GuillaumeBAECHLER** [comment]
I was having the same issue and using ```forceRefresh:true``` didn't seem to work. I ended up only refreshing the id_token by calling ```acquireTokenSilent``` and passing the clientId as the only scope in the Authentication Parameters.

**Comment by arunprasathv** [comment]
I did the same too.

**Comment by sameerag** [comment]
This could be a bug in the implicit flow. We do not necessarily set response_type=id_token_token when forceRefresh: true is set. We will check that part of the code, and if it is a bug as I suspect, will add the PR here.

**Comment by technical-boy** [comment]
@navzam we're investigating this issue and should have a fix soon. For now, a workaround would be to make separate acquireTokenSilent calls (one to get an access token and one to get an Id token). As of 1.3.0 you can use the ssoSilent() API to more easily acquire an Id token. 

```js
let idTokenResult = await myMSALObj.ssoSilent({
  loginHint: USER_UPN,
});

let authTokenResult = await myMSALObj.acquireTokenSilent({
  scopes: ['user.read']
});
```

**Comment by technical-boy** [comment]
This issue is now being addressed in #2022 and will be solved and closed together with that PR soon.

**Comment by navzam** [comment]
@technical-boy Awesome, thanks for the update. If I only need an ID token, is the recommendation still to use `ssoSilent()`?

**Comment by technical-boy** [comment]
Hi @navzam . Yes, until #2022 is merged, the previous recommendation of acquiring an ID Token through ssoSilent and then an access token through a separate acquireTokenSilent call is the most straight forward workaround!

**Comment by tnorling** [comment]
This was addressed in version 1.4.0. You can read more about the new response type logic [here](https://github.com/AzureAD/microsoft-authentication-library-for-js/blob/dev/lib/msal-core/docs/response-types.md). A separate bug when requesting **only** and idToken from `acquireTokenSilent` is being tracked in #2185
